docassemble/ALToolbox/al_income.py

```diff
@@ -189,45 +236,47 @@ def total(self, times_per_year=1, source=None, owner=None):
 
 class ALJob(ALIncome):
     """
-    Represents a job that may be hourly or pay-period based. The job is not
-    itemized. Can be stored in an ALJobList.
+    Represents a single job that may be hourly or pay-period based.
+
+    The job can have a net and gross income figure, but it does not represent
+    individual items like wages, tips or deductions that may appear on a paycheck--the
+    user must enter the total amount for "net" and "gross" income for a given period.
+
+    Can be stored in an ALJobList.
 
     Attributes:
-    .value { num } A number representing an amount of money accumulated during
+    .value {float | Decimal} A number representing an amount of money accumulated during
         the `times_per_year` of this income.
-    .times_per_year {str | num} Represents a number of the annual frequency of
+    .times_per_year {float} Represents a number of the annual frequency of
         the value. E.g. 12 for a monthly value.
-    .hours_per_period {num} (Optional) The number of hours during the annual
+    .is_hourly {bool} (Optional): Whether the gross total should be calculated based on hours
+        worked per week
+    .hours_per_period {float} (Optional) The number of hours during the annual
         frequency of this job. E.g. if the annual frequency is 52 (weekly), the
         hours per week might be 50. That is, 50 hours per week.
-    .net {num} (Optional) The net that the job makes during its pay period -
+    .net {float} (Optional) The net that the job makes during its pay period -
         its annual frequency. E.g. if the annual frequency is 52 (weekly), net
         is the total amount made for one week.
     .employer {str} (Optional) A single string for an employer's full name
     .employer_address {str} (Optional) A single string for employer's address
     .employer_phone {str} (Optional) An employer's phone number
     """
 
-    def total(self, times_per_year=1, source=None):
+    def gross_total(self, times_per_year: float = 1) -> Decimal:
```

Comment:
Looks like I totally got this wrong. I'm a bit confused about how `super` works, so I'm not sure if that needs to be used here with `self.total()`. That's the only question I have here, though.

Reply:
`super()` is only needed if we want to use the parent class and the child class overrides a method by the same name.